tests/test_ndarrays_regression.py

```diff
@@ -0,0 +1,599 @@
+        ndarrays_regression.check({"data1": data1})
+    obtained_error_msg = str(excinfo.value)
+    expected = "\n".join(
+        [
+            "Values are not sufficiently close.",
+            "To update values, use --force-regen option.",
+        ]
+    )
+    assert expected in obtained_error_msg
+    expected = "\n".join(
+        [
+            "data1:",
+            "  Shape: (3,)",
+            "  Number of differences: 2 / 3 (66.7%)",
+            "  Individual errors:",
+            "          Index              Obtained              Expected            Difference",
+            "              0                  True                 False                      ",
+            "              1                  True                 False                      ",
+        ]
+    )
+    assert expected in obtained_error_msg
+
+
+def test_complex_array(ndarrays_regression, no_regen):
+    # Correct data
+    data1 = np.array([3.0 + 2.5j, -0.5, -1.879j])
+    ndarrays_regression.check({"data1": data1})
+
+    # Data with errors
+    data1 = np.array([3.0 + 2.5j, 0.5, -1.879])
+    with pytest.raises(AssertionError) as excinfo:
+        ndarrays_regression.check({"data1": data1})
+    obtained_error_msg = str(excinfo.value)
+    expected = "\n".join(
+        [
+            "data1:",
+            "  Shape: (3,)",
+            "  Number of differences: 2 / 3 (66.7%)",
+            "  Statistics are computed for differing elements only.",
+            "  Stats for abs(obtained - expected):",
+            "    Max:     2.6573072836990455",
+            "    Mean:    1.8286536418495227",
+            "    Median:  1.8286536418495227",
+            "  Stats for abs(obtained - expected) / abs(expected):",
+            "    Max:     2.0",
+            "    Mean:    1.7071067811865475",
+            "    Median:  1.7071067811865475",
+            "  Individual errors:",
+            "          Index              Obtained              Expected            Difference",
+            "              1              (0.5+0j)             (-0.5+0j)                (1+0j)",
+            "              2           (-1.879+0j)           (-0-1.879j)       (-1.879+1.879j)",
+        ]
+    )
+
+
+def test_arrays_of_same_size_1d(ndarrays_regression):
+    data = {
+        "hello": np.zeros((1,), dtype=int),
+        "world": np.zeros((1,), dtype=int),
+    }
+    ndarrays_regression.check(data)
+
+
+def test_arrays_with_different_sizes_1d(ndarrays_regression, no_regen):
+    data = {"data1": np.ones(11, dtype=np.float64)}
+    ndarrays_regression.check(data)
+
+    # Original NPY file contains 11 elements.
+    data = {"data1": np.ones(10, dtype=np.float64)}
+    expected = re.escape(
+        "Shapes are not the same.\nKey: data1\nObtained: (10,)\nExpected: (11,)\n"
+    )
+    with pytest.raises(AssertionError, match=expected):
+        ndarrays_regression.check(data)
+
+
+def test_arrays_of_same_shape(ndarrays_regression):
+    same_size_int_arrays = {
```

Comment:
I believe you meant "same_shape" in the variable name? (Since the sizes are different)

Reply:
I missed that one. There is not so much use in giving the dictionary an elaborate name. Just `data` would be good enough because there is no potential for confusion with other dictionaries.